app/screens/home/recent_mentions/index.tsx

```diff
@@ -1,24 +1,110 @@
+import withObservables from '@nozbe/with-observables';
+import {useIsFocused, useNavigation} from '@react-navigation/native';
+import compose from 'lodash/fp/compose';
+import groupBy from 'lodash/groupBy';
+import moment from 'moment-timezone';
+import React, {useCallback, useState, useEffect, useMemo} from 'react';
+import {useIntl} from 'react-intl';
+import {StyleSheet, View, ActivityIndicator, SectionList, SectionListRenderItemInfo} from 'react-native';
 import Animated, {useAnimatedStyle, withTiming} from 'react-native-reanimated';
 import {SafeAreaView} from 'react-native-safe-area-context';
+import {of as of$, map as map$} from 'rxjs';
+import {catchError, switchMap} from 'rxjs/operators';
 
-const RecentMentionsScreen = () => {
-    const route = useRoute();
-    const params = route.params! as {direction: string};
-    const toLeft = params.direction === 'left';
-    const [start, setStart] = useState(false);
+import {getRecentMentions} from '@actions/remote/search';
+import DateSeparator from '@app/components/post_list/date_separator';
+import {MM_TABLES, SYSTEM_IDENTIFIERS} from '@app/constants/database';
+import {SystemModel, UserModel} from '@app/database/models/server';
+import NavigationHeader from '@components/navigation_header';
+import {LARGE_HEADER_PROGRESSVIEW_OFFSET} from '@constants/view';
+import {useServerUrl} from '@context/server_url';
+import {withTheme} from '@context/theme';
+import {useCollapsibleHeader} from '@hooks/header';
 
-    const onLayout = useCallback(() => {
-        setStart(true);
-    }, []);
+import Mention from './components/mention';
+
+import type {WithDatabaseArgs} from '@typings/database/database';
+import type PostModel from '@typings/database/models/servers/post';
+
+const AnimatedSectionList = Animated.createAnimatedComponent(SectionList);
+
+const {USER, SYSTEM, POST} = MM_TABLES.SERVER;
+
+type Props = {
+    mentions: PostModel[];
+    theme: Theme;
+    currentUser: UserModel;
+}
+
+type Section = {
+    key: string;
+    title: string;
+    data: PostModel[];
+    keyExtractor: (item: PostModel) => string;
+}
+
+const styles = StyleSheet.create({
+    loading: {
+        alignItems: 'center',
+        flex: 1,
+        justifyContent: 'center',
+    },
+});
+
+function getSections(items: PostModel[]) {
+    const groups = groupBy(items, (mention) => {
+        return moment(mention.createAt).startOf('day').format();
+    });
+
+    const sections: Section[] = Object.keys(groups).map((key) => ({
+        key: String(key),
+        title: String(key),
+        data: groups[key],
+        keyExtractor: (item: PostModel) => item.id,
+    }));
+
+    return sections.sort((a, b) => (a.title > b.title ? -1 : 1));
+}
+
+const RecentMentionsScreen = ({mentions, theme, currentUser}: Props) => {
+    const {formatMessage} = useIntl();
+    const [refreshing, setRefreshing] = useState(false);
+    const [loading, setLoading] = useState(false);
+    const searchScreenIndex = 1;
```

Comment:
It seems we only use this value on one place. Any objection to remove it and just use 1 in its place?

Reply:
The truth is that I copied that from the example @enahum made for the Search screen, but I don't truly understand the meaning of it, is it to animate screen slide from left or right? 
could we possibly make it a HOC to re-use?

Reply:
@koox00 This LGTM. Feel free to mark it as resolved if you already discussed this with enahum.